game/game.gd

```diff
@@ -1,12 +1,24 @@
 extends Node2D
 
-signal on_power_change(new_power_state: bool)
-
+@export var generator_power:int = 100
+signal on_generator_power_change(new_level: int)
+signal on_power_changed(power_state: bool)
 func _ready():
 	break_power(3)
-
+	
 func break_power(time: int):
 	await get_tree().create_timer(time).timeout
 	var breakers = get_tree().get_first_node_in_group('Breakers')
 	breakers.break_power(6)
 	# break_power(60)
+
+func use_generator_power(power: int):
+	generator_power -= power
+	on_generator_power_change.emit(generator_power)
```

Comment:
I would put the check for `<= 0` before this so it doesn't emit a power level value below 0. (idk how other code handles this value, so it might be unnecessary but idk)